### ISSUE TITLE
Telemetry on system required

### ISSUE THREAD

**Issue by PwL95** [FritzAndFriends/TagzApp#136]
Any type of logging or telemetry is required on the system to diagnose issues easier/ track system behavior

**Comment by csharpfritz** [comment]
Yes, let's get some telemetry flowing from the application.  I'd like to rename this issue to "Add OpenTelemetry and initial logging" and we should identify specific features that we would like to deliver telemetry data points.

Thoughts?

**Comment by napalm684** [comment]
I would like to take on implementing instrumentation for the solution with OTEL. As mentioned on stream I have been working on this at my day job and would like to continue learning as much as I can. I am thinking I can get the app instrumented with Traces, metrics and OTEL logging support and we can go from there on what to instrument.

I have also been experimenting with using Grafana, Loki, Tempo, Prometheus for visualization/querying of those observability items.

**Comment by csharpfritz** [comment]
> I would like to take on implementing instrumentation for the solution with OTEL. 

Let's talk through what you want to add and how we can configure this easily so that the OTEL capabilities can be enabled / disabled in different deployments.

**Comment by napalm684** [comment]
Yup makes sense. I wanted to add service collection extensions to setup OTEl with the ability to configure different exporters (ie azure monitor, zipkin, otlp, Prometheus, etc.)

**Comment by csharpfritz** [comment]
> Yup makes sense. I wanted to add service collection extensions to setup OTEl with the ability to configure different exporters (ie azure monitor, zipkin, otlp, Prometheus, etc.)

Let's hide this behind a feature flag so that we don't add the OTEL features when running in Desktop mode when this content is lifted and added to a MAUI Hybrid application.

The Blazor migration is nearing completion, and I would REALLY like to see the providers able to log information about their connection state and telemetry

**Comment by napalm684** [comment]
So just to ease things along I might wait for your blazor migration to be complete.
